cmd/access/node_builder/access_node_builder.go

```diff
@@ -722,6 +723,15 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 					if err != nil {
 						return nil, fmt.Errorf("could not load checkpoint file: %w", err)
 					}
+
+					// this is only supported when bootstrapping a new register db, otherwise the
+					// register db will throw exceptions when indexing an already indexed height.
+					if builder.forceResetIndexerHeight {
+						err = indexedBlockHeight.SetProcessedIndex(checkpointHeight)
```

Comment:
before resetting the index, it's better check the pebble database is empty, otherwise, the pebble DB will be built from a dirty state, and I'm not sure if the register index will behave correctly.

Reply:
OK, maybe we should add a check that if pebble DB is not bootstrapped, then the indexedBlockHeight should have no ProcessedIndex stored, otherwise it means the database is corrupted. 

So the database is corrupted if:
1. pebble DB is bootstrapped, but processed index is not set
2. pebble DB is not bootstrapped, but processed index is set

In case 1, we should probably add a check.
In case 2, if forceResetIndexerHeight == checkpointHeight, then we will reset the indexer height, otherwise, we should crash.

Reply:
1. could happen if the node crashed at some point during startup. we would have to check if pebble is bootstrapped and the register db's latest height is != root height and processed index is unset.

2. makes sense

Reply:
@zhangchiqing I added some checks and comments in [037a811](https://github.com/onflow/flow-go/pull/4943/commits/037a81195f70aaae364a694c0da0261ff6a54145). Still need to look into testing. Does this address your concern?